Marlin/src/pins/pins_ARMED.h

```diff
@@ -0,0 +1,190 @@
+#define Z_ENABLE_PIN       PD4
+#define Z_CS_PIN           PD5
+
+#define E0_STEP_PIN        PB5
+#define E0_DIR_PIN         PB6
+#define E0_CS_PIN          PB4
+
+#ifdef ARMED_V1_0
+  #define E0_ENABLE_PIN    PB3
+#else
+  #define E0_ENABLE_PIN    PC12
+#endif
+
+#ifdef ARMED_SWAP_X_E1
+  #define E1_STEP_PIN      PD3
+  #define E1_DIR_PIN       PD2
+  #define E1_ENABLE_PIN    PD0
+  #define E1_CS_PIN        PD1
+#else
+  #define E1_STEP_PIN      PE4
+  #define E1_DIR_PIN       PE2
+  #define E1_ENABLE_PIN    PE3
+  #define E1_CS_PIN        PE5
+#endif
+
+//
+// SPI
+//
+#define SCK_PIN            PA5
+#define MISO_PIN           PA6
+#define MOSI_PIN           PA7
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN         PC0   // Analog Input
+#define TEMP_1_PIN         PC1   // Analog Input
+#define TEMP_BED_PIN       PC2   // Analog Input
+
+//
+// Heaters / Fans
+//
+#define HEATER_0_PIN       PA1   // PWM pin
+#define HEATER_1_PIN       PA2   // PWM pin
+#define HEATER_BED_PIN     PA0   // PWM pin
+
+#define FAN_PIN            PC6   // PWM pin, Part cooling fan
+#define FAN1_PIN           PC7   // PWM pin, Extruder fan
+#define FAN2_PIN           PC8   // PWM pin, Controller fan
+
+//
+// Misc functions
+//
+#define SDSS               PE7
+#define SS_PIN             PE7
+#define LED_PIN            PB7   // Heart beat
+#define PS_ON_PIN          PA10
+#define KILL_PIN           PA8
+#define PWR_LOSS           PA4   // Power loss / nAC_FAULT
+
+//
+// LCD / Controller
+//
+#define SD_DETECT_PIN      PA15
+#define BEEPER_PIN         PC9
+#define LCD_PINS_RS        PE9
+#define LCD_PINS_ENABLE    PE8
+#define LCD_PINS_D4        PB12
+#define LCD_PINS_D5        PB13
+#define LCD_PINS_D6        PB14
+#define LCD_PINS_D7        PB15
+#define BTN_EN1            PC4
+#define BTN_EN2            PC5
+#define BTN_ENC            PC3
+
+//
+// Filament runout detection
+//
+#define FIL_RUNOUT_PIN     PA3
+
+//
+// Extension pins
+//
+#define EXT0_PIN           PB0
+#define EXT1_PIN           PB1
+#define EXT2_PIN           PB2
+#define EXT3_PIN           PD8
+#define EXT4_PIN           PD9
+#define EXT5_PIN           PD10
+#define EXT6_PIN           PD11
+#define EXT7_PIN           PD12
+#define EXT8_PIN           PB10
+#define EXT9_PIN           PB11
+
+//
+// ST7920 Delays
+//
+#define STM_DELAY_LONG      { __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); }
+#define STM_DELAY_SHORT     { __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); __asm__("nop\n\t"); }
+
+#ifndef ST7920_DELAY_1
+  #define ST7920_DELAY_1 { STM_DELAY_SHORT; STM_DELAY_SHORT; }
+#endif
+#ifndef ST7920_DELAY_2
+  #define ST7920_DELAY_2 { STM_DELAY_SHORT; }
+#endif
+#ifndef ST7920_DELAY_3
+  #define ST7920_DELAY_3 { STM_DELAY_LONG; STM_DELAY_LONG; STM_DELAY_LONG; STM_DELAY_LONG; STM_DELAY_LONG; STM_DELAY_LONG; }
+#endif
```

Comment:
We actually have `DELAY_NS` macros now that will do the right thing for each platform. If you know the exact timing of these delays, you should use `DELAY_NS` to replace these.